middle_end/flambda/types/structures/row_like.rec.ml

```diff
@@ -312,7 +355,9 @@ struct
       | Bottom ->
         match Tag.Map.get_singleton known_tags with
         | None -> None
-        | Some (tag, { maps_to; index }) ->
+        | Some (tag, { maps_to; index; env_extension = _ }) ->
+          (* If this is a singleton all the information from the
+             env_extension is already part of the environment *)
```

Comment:
This is only true because of implementation details. But I guess it's fine to ignore the extension in this case.

Reply:
I could document the invariant maybe. But I think that this is an important one (for performances): If there is only one case, the env_extension was already propagated.